### PR TITLE
test(file-upload): creates an example and test for file uploading

### PR DESCRIPTION
**Pull request by jmcdo29**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X ] Other... Please describe:
```

Samples and tests for file-upload

## What is the current behavior?

There's no sample for file-upload
Issue Number: N/A

## What is the new behavior?
There's a sample for file upload :smile_cat: 

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

File: sample/28-file-upload/src/sample.dto.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export class SampleDTO {
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/src/sample.dto.ts:1]: To remain consistent with the documentation, can we rename it to `SampleDto`?

File: sample/28-file-upload/src/app.service.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class AppService {
+
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/src/app.service.ts:5]:
```suggestion
```
To stay consistent with other examples, can we remove the unnecessary line?

File: sample/28-file-upload/src/app.controller.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Body, Controller, Get, Post, UploadedFile, UseInterceptors } from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Express } from 'express';
+import { AppService } from './app.service';
+import { SampleDTO } from './sample.dto';
+
+@Controller()
+export class AppController {
+
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/src/app.controller.ts:9]:
```suggestion
```
see below

File: sample/28-file-upload/src/app.controller.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(private readonly appService: AppService) {}
+
+  @Get()
+  sayHello(): { hello: string } {
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/src/app.controller.ts:13]: To stay consistent with other examples, can we remove the return type here?

File: sample/28-file-upload/src/app.controller.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { Express } from 'express';
+import { AppService } from './app.service';
+import { SampleDto } from './sample.dto';
+
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  @Get()
+  sayHello(): { hello: string } {
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/src/app.controller.ts:12]: @jmcdo29 return type is still there :D

File: sample/28-file-upload/e2e/app/app.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import * as request from 'supertest';
+import { AppModule } from '../../src/app.module';
+
+describe('E2E FileTest', () => {
+  let app: INestApplication;
+
+  beforeAll(async() => {
+    const modRef = await Test.createTestingModule({
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/e2e/app/app.e2e-spec.ts:12]:
```suggestion
    const moduleRef = await Test.createTestingModule({
```

File: sample/28-file-upload/e2e/app/app.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      imports: [AppModule],
+    }).compile();
+    app = modRef.createNestApplication();
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/e2e/app/app.e2e-spec.ts:15]:
```suggestion

    app = moduleRef.createNestApplication();
```

File: sample/28-file-upload/e2e/app/app.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
```
Comment (kamilmysliwiec) [inline sample/28-file-upload/e2e/app/app.e2e-spec.ts:21]: If we can move it to the end of the `describe()` block (to stay consistent with another PR), that would be perfect!

File: sample/28-file-upload/src/app.controller.ts (hunk 5 above)
Comment (jmcdo29) [reply sample/28-file-upload/src/app.controller.ts:12]: Old habits die really hard. I removed it from the `service` and forgot about the `controller`

File: sample/28-file-upload/e2e/app/app.e2e-spec.ts (hunk 8 above)
Comment (jmcdo29) [reply sample/28-file-upload/e2e/app/app.e2e-spec.ts]: There we go. I think I got everything lined up properly

**Comment by kamilmysliwiec** [comment]
Thanks @jmcdo29!

**Pull request merged**